clang/lib/CodeGen/CodeGenTypeCache.h

```diff
@@ -51,9 +51,11 @@ struct CodeGenTypeCache {
     llvm::IntegerType *PtrDiffTy;
   };
 
+  /// unqualified void*
+  llvm::PointerType *UnqualPtrTy;
```

Comment:
I'd argue we should just remove this variable, and the few cases that actually want AS0 can call PointerType::get(0) directly.

Reply:
I'm not opposed to this in principle; might have some additional benefits in that it forces people to knowingly opt for something via `PointerType::getUnqual` / `PointerType::get(some_as)`, rather than accidentally stumbling into it.